ecs-service/alb.tf

```diff
@@ -0,0 +1,46 @@
+resource "aws_lb_target_group" "this" {
+  name                 = "${var.environment}-${var.service_name}"
+  target_type          = local.target_type
+  port                 = var.container_port
+  protocol             = "HTTP"
+  vpc_id               = var.vpc_id
+  deregistration_delay = 120
+
+  health_check {
+    enabled = true
+    path    = var.health_check_path
+    port    = "traffic-port"
+  }
+}
+
+resource "aws_lb_listener_rule" "https" {
+  for_each = { for i, rule in var.listener_rules : i => rule }
```

Comment:
Good questions, may as well just use a `count` and `.index`. 
I did this because my mind was going around `for_each` and that requires a map.
Here it doesn't make sense too much, as the naming will be `aws_lb_listener_rule.https[<number>]` 
In some cases, for example in the SSM module we create a map of `name => object` which helps terraform removing an instance by key rather than by index (which can change if you change array elements)